tests/test_interp.py

```diff
@@ -155,6 +155,82 @@ def testLsstTextfile(self):
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(20, 12), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(20, 13), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(20, 14), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(20, 15), dimensions=lsst.geom.Extent2I(1, 1)))
+
+        # the normalize() method should replace the 10 defects above by just
+        # two defect boxes with adjusted dimensions
+        newDefects = defects.normalize()
+
+        boxesMeasured = []
+        for defect in newDefects:
+            boxesMeasured.append(defect.getBBox())
+
+        # The normalizing function should have created the following two boxes out
+        # of the individual 1-pixel defects from above
+        expectedDefects = [lsst.geom.Box2I(corner=lsst.geom.Point2I(15, 1),
+                                           dimensions=lsst.geom.Extent2I(1, 5)),
+                           lsst.geom.Box2I(corner=lsst.geom.Point2I(20, 11),
+                                           dimensions=lsst.geom.Extent2I(1, 5))]
+
+        self.assertEqual(len(expectedDefects), len(boxesMeasured))
+        for expDef, measDef in zip(expectedDefects, boxesMeasured):
+            self.assertEqual(expDef, measDef)
+
+        # Normalize two distinct sets of Defects and ensure they compare to the same thing
+        defects = algorithms.Defects()
+        # Set 1
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 1), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 2), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 3), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 4), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 5), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 6), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 7), dimensions=lsst.geom.Extent2I(1, 1)))
+        defects.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 8), dimensions=lsst.geom.Extent2I(1, 1)))
+
+        # Set 2
+        defects2 = algorithms.Defects()
+        defects2.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 1), dimensions=lsst.geom.Extent2I(1, 5)))
+        defects2.append(lsst.geom.Box2I(corner=lsst.geom.Point2I(25, 5), dimensions=lsst.geom.Extent2I(1, 4)))
+
+        # normalize both sets
+        newDefects = defects.normalize()
+        newDefects2 = defects2.normalize()
+
+        self.assertEqual(len(newDefects), len(newDefects2))
```

Comment:
You just need to `self.assertEqual(newDefects, newDefects2)`

No need to compare length and the individual defects since `__eq__` already does that for you.

Reply:
Thanks, I'll change that. This would compare `newDefects` and `newDefects2`, but I think I should still leave the lines below where each set is compared to the list  `expectedDefects`.